### PR TITLE
Fix Collection types

### LINKED ISSUE #76: Fix EnumSet
Currently, using an EnumSet results in code that doesn't compile. Please ensure that EnumSet works with all configuration items:
- nullable, mutable
- nullable, immutable
- never null, mutable
- never null, immutable

### PR DESCRIPTION
**Pull request by cbarlin**
Fixes #76. Need to create issue to make Eclipse Types follow this pattern, as well as add `Queue`.

## Summary by CodeRabbit

* **New Features**
  * Added comprehensive support for advanced collection types (e.g. EnumSet, TreeSet, LinkedList, Vector, Stack) in generated record builders, getters, setters, adders, mergers, and diff utilities.
  * Introduced new test records covering nullable and non-nullable combinations for a wide range of collection types.
  * Enhanced builder and utility generation for collections with improved nullability and immutability handling.

* **Bug Fixes**
  * Improved consistency and robustness in handling collection types, especially for enum-based collections and deep immutability.

* **Refactor**
  * Replaced multiple collection-specific visitor classes with a unified handler-based architecture for builder and utility code generation.
  * Standardised collection handling logic, reducing duplication and improving maintainability.

* **Chores**
  * Updated and expanded test coverage for odd and edge-case collection types.
  * Applied package-level nullness annotations for improved static analysis.
  * Cleaned up unused code and optimised internal constant definitions.

### CODE REVIEW

File: advanced-record-utils-processor/src/main/java/io/github/cbarlin/aru/impl/diff/utils/CollectonDiffCreation.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import io.micronaut.sourcegen.javapoet.TypeName;
+
+@ServiceProvider
+public final class CollectonDiffCreation extends DifferVisitor {
+
+    public CollectonDiffCreation() {
+        super(Claims.DIFFER_UTILS_COMPUTE_CHANGE);
+    }
+
+    private final Set<String> processedSpecs = HashSet.newHashSet(20);
```
[the remark this answers is not in the document]
Comment (cbarlin) [reply advanced-record-utils-processor/src/main/java/io/github/cbarlin/aru/impl/diff/utils/CollectonDiffCreation.java:29]: @coderabbitai `HasSet.newHashSet` is Java 21 style for the creation of `HashSet`s (it's not guava - this is the fourth time you've made that mistake...).

While it is Java 21 and up only, and we do eventually want to move to Java 17, if I change it to the `new HashSet<>()` style now before changing the compile target we'll get linter warnings everywhere telling us to use the new way of doing it.

File: advanced-record-utils-processor/src/main/java/io/github/cbarlin/aru/impl/types/collection/set/EnumSetCollectionHandler.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+@SuppressWarnings({"java:S1192"}) // Constant strings will not make this class clearer
+public final class EnumSetCollectionHandler extends SetCollectionHandler {
+
+    public EnumSetCollectionHandler() {
+        super(ENUM_SET, ENUM_SET);
+    }
+
+    public static void convertImmutable(final MethodSpec.Builder methodBuilder, final String fieldName, final String assignmentName, final TypeName innerTypeName) {
+        methodBuilder.addStatement("final $T<$T> $L = $T.copyOf($L)", SET, innerTypeName, assignmentName, SET, fieldName);
+    }
```
[the remark this answers is not in the document]
Comment (cbarlin) [reply advanced-record-utils-processor/src/main/java/io/github/cbarlin/aru/impl/types/collection/set/EnumSetCollectionHandler.java:31]: @coderabbitai The static methods are used elsewhere, so this is used to de-duplicate the logic

**Comment by cbarlin** [comment]
@coderabbitai resolve
